src/tests/integration/test_foundry_prove.py

```diff
@@ -66,28 +65,28 @@ def foundry_root(tmp_path_factory: TempPathFactory, worker_id: str) -> Path:
             run_process(['forge', 'build'], cwd=foundry_root)
 
             foundry_kompile(
-                foundry_root=foundry_root,
+                foundry=Foundry(foundry_root),
                 includes=(),
                 requires=[str(TEST_DATA_DIR / 'lemmas.k')],
                 imports=['LoopsTest:SUM-TO-N-INVARIANT'],
             )
 
     session_foundry_root = tmp_path_factory.mktemp('foundry')
     copy_tree(str(foundry_root), str(session_foundry_root))
-    return session_foundry_root
+    return Foundry(session_foundry_root)
 
 
-def test_foundry_kompile(foundry_root: Path, update_expected_output: bool, use_booster: bool) -> None:
+def test_foundry_kompile(foundry: Foundry, update_expected_output: bool, use_booster: bool) -> None:
     if use_booster:
         return
     # Then
     assert_or_update_k_output(
-        foundry_root / 'out/kompiled/foundry.k',
+        foundry._root / 'out/kompiled/foundry.k',
         TEST_DATA_DIR / 'foundry.k.expected',
         update=update_expected_output,
     )
     assert_or_update_k_output(
-        foundry_root / 'out/kompiled/contracts.k',
+        foundry._root / 'out/kompiled/contracts.k',
```

Comment:
```suggestion
        foundry.kompiled / 'contracts.k',
```

But introducing a new property `contracts_file` would be even better.